### PR TITLE
refactor: DRYify query code

### LINKED ISSUE #10: Refactor QueryService.js
There's a lot of duplicated code in this module.

### PR DESCRIPTION
**Pull request by stdavis**
Closes #10

### CODE REVIEW

File: src/services/QueryService.js [hunk 1]
```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
...
+  console.log('QueryService.queryReferenceTableAsync');
 
-  const response = await fetch(`${url}/query?${stringify(parameters)}`);
-  const responseJson = await response.json();
+  units = getDistinctHazardCodesFromUnits(units);
+  const where = `Hazard IN ('${Array.from(units).join('\',\'')}')`;
+  const outFields = 'Hazard,Text';
 
-  return responseJson.features.map(feature => feature.attributes);
+  return queryTable(config.urls.hazardReferenceTextTable, where, outFields);
```
Comment (steveoh) [inline src/services/QueryService.js]: This async modifier can be removed or you'll need to await something in the method.

```suggestion
export const queryReferenceTableAsync = (units) => {
```

```suggestion
  return await queryTable(config.urls.hazardReferenceTextTable, where, outFields);
```

You are awaiting all of these in a promise in app.js so I think returning the promise is fine?

File: src/services/QueryService.js [hunk 2]
```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
...
 };
 
 export const queryIntroTextAsync = async (units) => {
```
Comment (steveoh) [inline src/services/QueryService.js]:
```suggestion
export const queryIntroTextAsync = (units) => {
```

File: src/services/QueryService.js [hunk 3]
```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
...
-  const responseJson = await response.json();
+  units = getDistinctHazardCodesFromUnits(units);
+  const where = `Hazard IN ('${Array.from(units).join('\',\'')}')`;
+  const outFields = 'Hazard,Text';
 
-  return responseJson.features.map(feature => feature.attributes);
+  return queryTable(config.urls.hazardIntroTextTable, where, outFields);
 };
 
 export const queryGroupingAsync = async (units) => {
```
Comment (steveoh) [inline src/services/QueryService.js]:
```suggestion
export const queryGroupingAsync = (units) => {
```

File: src/services/QueryService.js [hunk 4]
```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
...
-  };
-
-  const response = await fetch(`${url}/query?${stringify(parameters)}`);
-  const responseJson = await response.json();
-
-  return responseJson.features.map(feature => feature.attributes);
+  return queryTable(config.urls.hazardGroupingsTable, where, outFields);
 };
 
 export const queryGroupTextAsync = async (groups) => {
```
Comment (steveoh) [inline src/services/QueryService.js]:
```suggestion
export const queryGroupTextAsync = (groups) => {
```

File: src/services/QueryService.js [hunk 5]
```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
...
-  };
-
-  const response = await fetch(`${url}/query?${stringify(parameters)}`);
-  const responseJson = await response.json();
-
-  return responseJson.features.map(feature => feature.attributes);
+  return queryTable(config.urls.hazardGroupTextTable, where, outFields);
 };
 
 export const queryReportTextTableAsync = async () => {
```
Comment (steveoh) [inline src/services/QueryService.js]:
```suggestion
export const queryReportTextTableAsync = () => {
```

File: src/services/QueryService.js [hunk 6]
```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
...
-  };
-
-  const response = await fetch(`${url}/query?${stringify(parameters)}`);
-  const responseJson = await response.json();
-
-  return responseJson.features.map(feature => feature.attributes);
+  return queryTable(config.urls.reportTextTable, where, outFields);
 };
 
 export const queryOtherDataTable = async () => {
```
Comment (steveoh) [inline src/services/QueryService.js]:
```suggestion
export const queryOtherDataTable = () => {
```

File: src/services/QueryService.js [hunk 7]
```diff
@@ -55,130 +45,68 @@ export const queryHazardUnitTableAsync = async (units) => {
...
-  const url = config.urls.hazardReferenceTextTable;
+export const queryHazardUnitTableAsync = async (units) => {
+  console.log('QueryService.queryHazardUnitTableAsync');
 
-  units = new Set(units.map(unit => getHazardCodeFromUnitCode(unit)));
+  const where = `HazardUnit IN ('${units.join('\',\'')}')`;
+  const outFields = 'HazardName,HazardUnit,HowToUse,Description';
 
-  const whereClause = `Hazard IN ('${Array.from(units).join('\',\'')}')`;
+  return await queryTable(config.urls.hazardUnitTextTable, where, outFields);
```
Comment (steveoh) [inline src/services/QueryService.js]: this is the only method you await the query table, is that on purpose?

Comment (stdavis) [reply src/services/QueryService.js]: No, that was an oversight.

**Comment by stdavis** [comment]
Great catch. I removed all of the asyncs in one commit rather than accepting all of your suggestions one at a time.

**Pull request merged**